internal/tasks/device_render.go

```diff
@@ -66,29 +70,54 @@ func (t *DeviceRenderLogic) RenderDevice(ctx context.Context) error {
 	emptyIgnitionConfig, _, _ := config_latest.ParseCompatibleVersion([]byte("{\"ignition\": {\"version\": \"3.4.0\"}"))
 	t.renderedConfig = &emptyIgnitionConfig
 
+	invalidConfigs := []string{}
 	if device.Spec != nil && device.Spec.Config != nil {
 		for i := range *device.Spec.Config {
 			configItem := (*device.Spec.Config)[i]
-			err := t.handleConfigItem(ctx, &configItem)
+			name, err := t.handleConfigItem(ctx, &configItem)
 			if err != nil {
-				return t.setStatus(ctx, err)
+				invalidConfigs = append(invalidConfigs, name)
 			}
 		}
 	}
 
+	repoRefsErr := t.store.Device().OverwriteRepositoryRefs(ctx, t.resourceRef.OrgID, *device.Metadata.Name, t.repoNames...)
+
+	if len(invalidConfigs) != 0 {
+		configurationStr := "configuration"
+		if len(invalidConfigs) > 1 {
+			configurationStr += "s"
+		}
+		err = fmt.Errorf("device has %d invalid %s: %s", len(invalidConfigs), configurationStr, strings.Join(invalidConfigs, ", "))
+		return t.setStatus(ctx, err)
+	}
+
+	if repoRefsErr != nil {
+		return t.setStatus(ctx, repoRefsErr)
+	}
+
 	configjson, err := json.Marshal(t.renderedConfig)
 	if err != nil {
 		return t.setStatus(ctx, err)
 	}
 
+	if device.Metadata.Owner == nil || *device.Metadata.Owner == "" {
+		// The standalone device is valid, set the many-to-may relationship with the repos
+		err = t.store.Device().OverwriteRepositoryRefs(ctx, t.resourceRef.OrgID, *device.Metadata.Name, t.repoNames...)
+		if err != nil {
+			return fmt.Errorf("setting repository references: %w", err)
+		}
+	}
+
 	err = t.store.Device().UpdateRendered(ctx, t.resourceRef.OrgID, t.resourceRef.Name, string(configjson))
 	return t.setStatus(ctx, err)
 }
 
-func (t *DeviceRenderLogic) handleConfigItem(ctx context.Context, configItem *api.DeviceSpec_Config_Item) error {
+func (t *DeviceRenderLogic) handleConfigItem(ctx context.Context, configItem *api.DeviceSpec_Config_Item) (string, error) {
+
 	disc, err := configItem.Discriminator()
 	if err != nil {
-		return fmt.Errorf("failed getting discriminator: %w", err)
+		return unknownName, fmt.Errorf("failed getting discriminator: %w", err)
```

Comment:
Could we do this another way generally speaking idiomatic go assumes the value returned from a method is not used. If this error means something then wrap the err.

```suggestion
		return fmt.Errorf("failed getting discriminator %w: %w", ErrUnknownName, err)